app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -6,23 +6,76 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
   #
   ###
 
-	def google_oauth2
-		# You need to implement the method below in your model (e.g. app/models/user.rb)
-		@user = User.from_omniauth(request.env["omniauth.auth"])
-
-		if @user.persisted?
-			@user.update(authentication_token: Devise.friendly_token(32))
-			@user.generate_access_token
-			# update a user's FireCloud status
-			@user.delay.update_firecloud_status
-			sign_in(@user)
-			if TosAcceptance.accepted?(@user)
-				redirect_to request.env['omniauth.origin'] || site_path
-			else
-				redirect_to accept_tos_path(@user.id)
-			end
-		else
-			redirect_to new_user_session_path
-		end
-	end
+
+
+  # Merges unauth’d and auth’d user identities in Mixpanel via Bard
+  #
+  # Bard is a DSP service that mediates writes to Mixpanel. To enable tracking
+  # users when they are signed in and not, Bard provides an endpoint
+  # `/api/identify` to merge identities.  In SCP's case, the anonynmous ID
+  # (`anonId`) is a random UUIDv4 string set as a cookie for all users --
+  # auth'd or not -- upon visiting SCP.
+
+  # This call links that anonId to the user's bearer token used by DSP's Sam
+  # service.  That bearer token is in turn linked to a deidentified
+  # "distinct ID" used to track users across auth states in Mixpanel.
+  def merge_identities
```

Comment:
I would make this method static and accept the user as an argument.  That will help if we need to write, e.g. a migration that merges a bunch of old users